src/loc/LocItem.tsx

```diff
@@ -30,29 +30,259 @@ import { ContributionMode } from "./types";
 import ReviewStatusCell from "./ReviewStatusCell";
 import HelpTooltip from "src/components/helptooltip/HelpTooltip";
 
-export type LocItemType = 'Data' | 'Document' | 'Linked LOC'
-
-export interface LocItem {
-    hash?: Hash,
-    name: string | undefined,
-    value: string | undefined,
-    timestamp: string | null,
-    type: LocItemType,
-    submitter: ValidAccountId | undefined,
-    status: ItemStatus,
-    nature?: string,
-    target?: UUID,
-    newItem: boolean,
-    linkDetailsPath?: string,
-    template: boolean,
-    isSet?: boolean,
-    size?: bigint,
-    fees?: Fees,
-    storageFeePaidBy?: string;
-    reviewedOn?: string;
-    rejectReason?: string;
-    acknowledgedByOwner?: boolean;
-    acknowledgedByVerifiedIssuer?: boolean;
+export type LocItemType = 'Data' | 'Document' | 'Linked LOC';
+
+export interface LinkData {
+    readonly linkedLoc: LocData;
+    readonly linkDetailsPath: string;
+    readonly nature: string;
+}
+
+export interface MetadataData {
+    readonly name?: string;
+    readonly nameHash: Hash;
+    readonly value: string;
+}
+
+export interface FileData {
+    readonly hash: Hash;
+    readonly fileName: string;
+    readonly nature: string;
+    readonly size: bigint;
+    readonly storageFeePaidBy: string;
+}
+
+export interface CommonData {
+    readonly timestamp: string | null;
+    readonly type: LocItemType;
+    readonly submitter?: ValidAccountId;
+    readonly status: ItemStatus;
+    readonly newItem: boolean;
+    readonly template: boolean;
+    readonly isSet?: boolean;
+    readonly fees?: Fees;
+    readonly reviewedOn?: string;
+    readonly rejectReason?: string;
+    readonly acknowledgedByOwner?: boolean;
+    readonly acknowledgedByVerifiedIssuer?: boolean;
+}
+
+export abstract class LocItem implements CommonData {
```

Comment:
Did you try to use generics here ? Seems like a typical case, this would avoid to reimplement `metadataData()` & co where useless. But that may make things more difficult at call site.
```suggestion
export abstract class LocItem<T> implements CommonData {
  abstract data(): T;
}
...
export class MetadataItem extends LocItem<MetadataData> {}
```

Reply:
Good point. As discussed, you would need an expression like `item instanceof MetadataItem && item.data().value` for the compiler to know what the return type of `data()` is, which is not necessarily less readable than `item.type === "Data" && item.metadataData().value`. I let you try.